### PR TITLE
Fixes the distance issue between the legend and the horizontal bar chart (Fixes #2138)

### LINKED ISSUE #2138: Distance between chart legend and the horizontal bar chart is to big in case of rotated labels
I rechecked the issue with Charts 3.0.1. The behavior is the same. See images below.

```Swift
chart.xAxis.labelRotationAngle       = -90.0           // rotate label
chart.xAxis.centerAxisLabelsEnabled  = true
```

bildschirmfoto 2017-02-04 um 22 16 43

```Swift
chart.xAxis.centerAxisLabelsEnabled  = true
```

bildschirmfoto 2017-02-04 um 22 19 45

See also my old issue #1360

### PR DESCRIPTION
**Pull request by SvenMuc**
Override the function calculateLegendOffsets in HorizontalBarChartView
in order to select the correct axis for distance calculation. In case
of horizontal chart the x-axis is described the the rightAxis (bottom)
or the leftAxis(top).

### CODE REVIEW

**Comment by SvenMuc** [comment]
Seems to be an timeout issue:

```
1201.47s$ travis_wait carthage bootstrap --platform $PLATFORM
Still running (20 of 20): carthage bootstrap --platform iOS
Timeout (20 minutes) reached. Terminating "carthage bootstrap --platform iOS"
The command carthage bootstrap --platform iOS exited with 137.
Log:
*** Cloning ios-snapshot-test-case
*** Cloning realm-cocoa
*** Skipped downloading ios-snapshot-test-case.framework binary due to the error:
	"Bad credentials"
*** Checking out ios-snapshot-test-case at "da629211c17a4c507e2e866e8a19ed3122af770b"
*** Skipped downloading realm-cocoa.framework binary due to the error:
	"Bad credentials"
*** Checking out realm-cocoa at "v2.1.1"
*** xcodebuild output can be found in /var/folders/bb/n7t3rs157850byt_jfdcq9k80000gn/T/carthage-xcodebuild.f9E0AX.log
*** Building scheme "FBSnapshotTestCase iOS" in FBSnapshotTestCase.xcworkspace
*** Building scheme "Realm" in Realm.xcworkspace
*** Building scheme "RealmSwift" in Realm.xcworkspace
The command "travis_wait carthage bootstrap --platform $PLATFORM" failed and exited with 137 during .
```

**Comment by SvenMuc** [comment]
Any news when it will be integrated into the official release. It's an important fix for my actual app. Otherwise I have to modify the code with every Charts update manually. Thanks for some feedback.

**Comment by SvenMuc** [comment]
Test fix with v3.0.4. 

@danielgindi @liuxuan30 @petester42 
When are you planning to merge this fix into the official release?

**Comment by liuxuan30** [comment]
@jjatie how about this? Either 3.x or 4.0 is fine. I think for bug fixes, master is usually the branch to merge.

**Comment by liuxuan30** [comment]
@SvenMuc are you able to pull master and trigger a new CI? we recently add some new tests.

**Comment by SvenMuc** [comment]
@liuxuan30 Do you mean to merge the master into this pull request? I can try it.

**Comment by jjatie** [comment]
@SvenMuc Yes. We recently added new tests that need to pass before this can be merged. Pulling master into this branch will fix that. Also note the requested changes.

**Comment by SvenMuc** [comment]
@jjatie Ok, I merged the master into this branch. Your comments I'm going to check tomorrow evening.

File: Source/Charts/Charts/HorizontalBarChartView.swift [hunk 1]
```diff
@@ -34,6 +34,64 @@ open class HorizontalBarChartView: BarChartView
         self.highlighter = HorizontalBarHighlighter(chart: self)
     }
     
+    internal override func calculateLegendOffsets(offsetLeft: inout CGFloat, offsetTop: inout CGFloat, offsetRight: inout CGFloat, offsetBottom: inout CGFloat)
+    {
+        // setup offsets for legend
+        if _legend !== nil && _legend.isEnabled && !_legend.drawInside
```
[the remark this answers is not in the document]
Comment (SvenMuc) [reply Source/Charts/Charts/HorizontalBarChartView.swift]: Done

File: Source/Charts/Charts/HorizontalBarChartView.swift [hunk 2]
```diff
@@ -34,6 +34,64 @@ open class HorizontalBarChartView: BarChartView
...
+        {
+            switch _legend.orientation
```
[the remark this answers is not in the document]
Comment (SvenMuc) [reply Source/Charts/Charts/HorizontalBarChartView.swift]: Done

**Comment by jjatie** [comment]
@SvenMuc Travis CI has had some issues the last few days. When they're up and running for us, we will get this merged.

**Pull request merged**